solver/src/solver/http_solver.rs

```diff
@@ -364,14 +367,19 @@ impl HttpSolver {
             })
             .collect();
 
-        let price_estimates: HashMap<H160, f64> = queries
+        let mut new_price_estimates: HashMap<H160, f64> = queries
             .iter()
-            .zip(price_estimates)
+            .zip(new_price_estimates)
             .filter_map(|(query, estimate)| {
                 let price = estimate.ok()?.price_in_sell_token_f64(query);
                 Some((query.buy_token, price))
             })
             .collect();
+        new_price_estimates.extend(
+            price_estimates
+                .into_iter()
+                .filter_map(|(token, price)| Some((token, price.to_f64()?))),
+        );
```

Comment:
Nick mentioned that the http solver shouldn't need new price estimates. I think it does because:
- driver only estimates all tokens appearing in user limit orders
- http solver estimates all tokens even from liquidity
- liquidity contains base tokens that we haven't price estimated yet in the driver

Reply:
> http solver estimates all tokens even from liquidity

I imagine that the `set all tokens from liquidity` == `set all tokens in user limit orders` + `set of all base token`. So estimating prices from any token included in liquidity should not need to consider any new pools than are already there.

Reply:
The pools are there but the price estimate isn't. So we are going to make calls to the price estimator but the pools it uses should be cached.

Reply:
Right! I think I may have not been super clear with what I meant: the HTTP solver can just use the `Batch` input as all pools available for baseline price estimation instead of needing to rely on the pool cache. 

Its also fine to just rely on the pool cache though :stuck_out_tongue:.